x-pack/plugins/security_solution/public/overview/containers/overview_cti_links/get_event_counts.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useState, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { MatrixHistogramType } from '../../../../common/search_strategy/security_solution';
+import { useKibana } from '../../../common/lib/kibana';
+import { ThreatIntelLinkPanelProps } from '../../components/overview_cti_links';
+import { convertToBuildEsQuery } from '../../../common/lib/keury';
+import { esQuery, Filter } from '../../../../../../../src/plugins/data/public';
+import { useMatrixHistogram } from '../../../common/containers/matrix_histogram';
+import { EVENT_DATASET } from '../../../../common/cti/constants';
+
+export const ID = 'ctiEventCountQuery';
+
+const ctiEventsFilter: Filter = {
+  meta: {
+    alias: null,
+    disabled: false,
+    key: 'event.dataset',
+    negate: false,
+    params: {
+      query: 'file',
+    },
+    type: 'phrase',
+  },
+  query: {
+    match_phrase: { 'event.module': 'threatintel' },
+  },
+};
+
+export const useCTIEventCounts = ({
+  deleteQuery,
+  filters,
+  from,
+  indexNames,
+  indexPattern,
+  query,
+  setQuery,
+  to,
+}: ThreatIntelLinkPanelProps) => {
+  const [isInitialLoading, setIsInitialLoading] = useState(true);
+  const { uiSettings } = useKibana().services;
+
+  useEffect(() => {
+    return () => {
+      if (deleteQuery) {
+        deleteQuery({ id: ID });
+      }
+    };
+  }, [deleteQuery]);
+
+  const matrixHistogramRequest = useMemo(
+    () => ({
+      endDate: to,
+      errorMessage: i18n.translate('xpack.securitySolution.overview.errorFetchingEvents', {
+        defaultMessage: 'Error fetching events',
+      }),
+      filterQuery: convertToBuildEsQuery({
+        config: esQuery.getEsQueryConfig(uiSettings),
+        indexPattern,
+        queries: [query],
+        filters: [...filters, ctiEventsFilter],
+      }),
+      histogramType: MatrixHistogramType.events,
+      indexNames,
+      stackByField: EVENT_DATASET,
+      startDate: from,
+    }),
+    [filters, from, indexPattern, uiSettings, query, to, indexNames]
+  );
+
+  const [loading, { data, inspect, totalCount, refetch }] = useMatrixHistogram(
+    matrixHistogramRequest
+  );
+
+  useEffect(() => {
+    if (!loading && !isInitialLoading) {
+      setQuery({ id: ID, inspect, loading, refetch });
+    }
+  }, [setQuery, inspect, loading, refetch, isInitialLoading, setIsInitialLoading]);
+
+  useEffect(() => {
+    if (isInitialLoading && data) {
+      setIsInitialLoading(false);
+    }
+  }, [isInitialLoading, data]);
+
+  const returnVal = {
+    eventCounts: data.reduce((acc, item) => {
+      if (item.y && item.g?.match('threatintel.')) {
```

Comment:
Instead of passing an empty query and checking the format of the dataset value (which may not always conform to this structure), I think it'll be more accurate and simpler to query instead for `event.kind: 'indicator'`, which should guarantee that we're only dealing with indicator data.